addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,193 @@
+                                .raise();
+                    }
+                }
+
+                // sending payloads in Headers.
+                for (String h : hpayloads) {
+                    String TMP_URI = SCHEMA + "://" + HOST + PATH;
+                    if (h.contains("X-Rewrite-URL")) {
+                        TMP_URI = SCHEMA + "://" + HOST + "/anything";
+                    } else if (h.contains("X-Original-URL")) {
+                        TMP_URI = SCHEMA + "://" + HOST + "/";
+                    } else if (h.contains("Referer")) {
+                        TMP_URI = SCHEMA + "://" + HOST + "/anything";
+                    }
+                    HttpMessage req_with_payload1 =
+                            new HttpMessage(new org.apache.commons.httpclient.URI(TMP_URI, true));
+                    req_with_payload1
+                            .getRequestHeader()
+                            .setHeader(h.split(":")[0], h.split(":")[1]);
+                    sendAndReceive(req_with_payload1);
+                    if (req_with_payload1.getResponseHeader().getStatusCode()
+                            == HttpStatusCode.OK) {
+                        System.out.println(req_with_payload1.getResponseHeader().getStatusCode());
+                        newAlert()
+                                .setConfidence(Alert.RISK_INFO)
+                                .setName(
+                                        Constant.messages.getString(
+                                                "ascanalpha.forbiddenBypass.name"))
+                                .setDescription(
+                                        Constant.messages.getString(
+                                                "ascanalpha.forbiddenBypass.desc"))
+                                .setOtherInfo(FULL_URL)
+                                .setMessage(req_with_payload1)
+                                .raise();
+                    }
+                }
+            }
+        } catch (IOException | URISyntaxException e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.MISC;
+    }
+
+    @Override
+    public String getSolution() {
+        return "";
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString("ascanalpha.forbiddenBypass.refs");
+    }
+
+    public List<Alert> getExampleAlerts() {
+        List<Alert> alerts = new ArrayList<Alert>();
+        Alert alert =
+                newAlert()
+                        .setConfidence(Alert.RISK_INFO)
+                        .setName(Constant.messages.getString("ascanalpha.forbiddenBypass.name"))
+                        .setDescription(
+                                Constant.messages.getString("ascanalpha.forbiddenBypass.desc"))
+                        .setReference(
+                                Constant.messages.getString("ascanalpha.forbiddenBypass.refs"))
```

Comment:
This should be extracted to a method to avoid duplicating here and in the scan method.

Reply:
I've added method for this please take a look at updated file.

Reply:
The method `getExampleAlerts()` is still duplicating the code. The PRs linked earlier show how to avoid that.